### PR TITLE
fix: ldap deps in `install-dependencies.sh`

### PR DESCRIPTION
**Pull request by leslie-tsang**
Signed-off-by: leslie <leslie.tsang@icloud.com>

### What this PR does / why we need it:

* use `openldap-devel` instead of `libldap2-dev`,  `libldap2-dev` don't exist on centos
* fix file lint in `t/plugin/ldap-auth.t`, add new line at the end of file

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: utils/install-dependencies.sh [hunk 1]
```diff
@@ -24,10 +24,9 @@ function install_dependencies_with_yum() {
     # add OpenResty source
     sudo yum install yum-utils
     sudo yum-config-manager --add-repo "https://openresty.org/package/${1}/openresty.repo"
-    sudo yum check-update
```
Comment (spacewander) [inline utils/install-dependencies.sh:27]: Why remove this line?

Comment (leslie-tsang) [reply utils/install-dependencies.sh:27]: When an update is available, `yum check-update` returns status code `100`, Which will break the execution.

Comment (spacewander) [inline utils/install-dependencies.sh:27]: `sudo yum check-update || true` would be better?

Comment (leslie-tsang) [reply utils/install-dependencies.sh:27]: IMO, `yum check-update` check whether there is a update or not would't bring any benefit for `install_dependencies_with_yum`.
`yum` don't need to update the repo cache after adding a new repo like `apt` does.
Shall we remove it ?

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**